### PR TITLE
Fix `get-details` and `install` for VersionFormat changes

### PR DESCRIPTION
**Pull request by superm1**
When these values are coming from the metadata in the CAB file, it's important to apply them to the checks used for these two commands.

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/hughsie/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-tool.c [hunk 1]
```diff
@@ -878,10 +878,23 @@ fu_util_install (FuUtilPrivate *priv, gchar **values, GError **error)
 
 			/* is this component valid for the device */
 			task = fu_install_task_new (device, component);
+			if (!fu_engine_check_requirements (priv->engine,
+							   task, priv->flags | FWUPD_INSTALL_FLAG_IGNORE_VERFMT,
```
Comment (hughsie) [inline src/fu-tool.c]: would FORCE work here as well?

File: src/fu-install-task.c [hunk 2]
```diff
@@ -334,12 +334,14 @@ fu_install_task_check_requirements (FuInstallTask *self,
 	}
 
 	/* check the version formats match if set in the release */
-	verfmts = xb_node_query (self->component,
-				 "custom/value[@key='LVFS::VersionFormat']",
-				 0, NULL);
-	if (verfmts != NULL) {
-		if (!fu_install_task_check_verfmt (self, verfmts, flags, error))
-			return FALSE;
+	if ((flags & FWUPD_INSTALL_FLAG_IGNORE_VERFMT) == 0) {
```
Comment (hughsie) [inline src/fu-install-task.c]: Is this also something we want to allow using FORCE?

File: libfwupd/fwupd-enums.h [hunk 3]
```diff
@@ -174,6 +174,7 @@ typedef guint64 FwupdReleaseFlags;
  * @FWUPD_INSTALL_FLAG_ALLOW_OLDER:		Allow downgrading firmware
  * @FWUPD_INSTALL_FLAG_FORCE:			Force the update even if not a good idea
  * @FWUPD_INSTALL_FLAG_NO_HISTORY:		Do not write to the history database
+ * @FWUPD_INSTALL_FLAG_IGNORE_VERFMT:	Ignore the version format when checking requirements
```
Comment (hughsie) [inline libfwupd/fwupd-enums.h]: indent, if we need this at all....

File: libfwupd/fwupd-client.c [hunk 4]
```diff
@@ -1045,6 +1045,10 @@ fwupd_client_install (FwupdClient *client,
 		g_variant_builder_add (&builder, "{sv}",
 				       "no-history", g_variant_new_boolean (TRUE));
 	}
+	if (install_flags & FWUPD_INSTALL_FLAG_IGNORE_VERFMT) {
```
Comment (hughsie) [inline libfwupd/fwupd-client.c]: i think this is internal use only, and no need to parse.

File: src/fu-tool.c (hunk 1 above)
Comment (superm1) [reply src/fu-tool.c]: Yes it does, adjusted to that much better thanks.

**Pull request merged**